app/integrations/google_workspace/google_directory.py

```diff
@@ -1,119 +1,103 @@
+        delegated_user_email = DEFAULT_DELEGATED_ADMIN_EMAIL
     scopes = ["https://www.googleapis.com/auth/admin.directory.user.readonly"]
-
-    service = get_google_service(
+    return execute_google_api_call(
         "admin",
         "directory_v1",
-        delegated_user_email=GOOGLE_DELEGATED_ADMIN_EMAIL,
-        scopes=scopes,
+        "users",
+        "get",
+        scopes,
+        delegated_user_email,
+        userKey=user_key,
     )
-    user = service.users().get(userKey=user_key).execute()
-    return user
 
 
 @handle_google_api_errors
-def list_users():
+def list_users(
+    delegated_user_email=None,
+    customer=None,
+):
     """List all users in the Google Workspace domain.
 
     Returns:
         list: A list of user objects.
-
-    Ref: https://developers.google.com/admin-sdk/directory/reference/rest/v1/users/list
     """
-
+    if not delegated_user_email:
+        delegated_user_email = DEFAULT_DELEGATED_ADMIN_EMAIL
+    if not customer:
+        customer = DEFAULT_GOOGLE_WORKSPACE_CUSTOMER_ID
     scopes = ["https://www.googleapis.com/auth/admin.directory.user.readonly"]
-    service = get_google_service(
+    return execute_google_api_call(
         "admin",
         "directory_v1",
-        delegated_user_email=GOOGLE_DELEGATED_ADMIN_EMAIL,
-        scopes=scopes,
+        "users",
+        "list",
+        scopes,
+        delegated_user_email,
+        paginate=True,
+        customer=customer,
+        maxResults=500,
+        orderBy="email",
     )
-    page_token = None
-    all_users = []
-    while True:
-        results = (
-            service.users()
-            .list(
-                customer=GOOGLE_WORKSPACE_CUSTOMER_ID,
-                maxResults=500,
-                orderBy="email",
-                pageToken=page_token,
-            )
-            .execute()
-        )
-        all_users.extend(results.get("users", []))
-        page_token = results.get("nextPageToken")
-        if not page_token:
-            break
-    return all_users
 
 
 @handle_google_api_errors
-def list_groups():
+def list_groups(
+    delegated_user_email=None,
+    customer=None,
+):
     """List all groups in the Google Workspace domain.
 
     Returns:
         list: A list of group objects.
 
     Ref: https://developers.google.com/admin-sdk/directory/reference/rest/v1/groups/list
     """
-
+    if not delegated_user_email:
+        delegated_user_email = DEFAULT_DELEGATED_ADMIN_EMAIL
+    if not customer:
+        customer = DEFAULT_GOOGLE_WORKSPACE_CUSTOMER_ID
     scopes = ["https://www.googleapis.com/auth/admin.directory.group.readonly"]
-
-    service = get_google_service(
+    return execute_google_api_call(
         "admin",
         "directory_v1",
-        delegated_user_email=GOOGLE_DELEGATED_ADMIN_EMAIL,
-        scopes=scopes,
+        "groups",
+        "list",
+        scopes,
+        delegated_user_email,
+        paginate=True,
+        customer=customer,
+        maxResults=100,
```

Comment:
Same about maxResults being 100. That's probably sufficient but original code has 500 .

Reply:
the max allowed seems to be 200 in the API reference so I'll update with that value!